### ISSUE TITLE
feat(drag-drop): prevent a item be removed when it into another container

### ISSUE THREAD

**Issue by Eve-Sama** [angular/components#20194]
#### Feature Description

CDK(V9) provide `[cdkDropListEnterPredicate]` to prevent a item into another container, provide `(cdkDropListDropped)` to determine how to do when user drop the item. 

I found when user drag the item into another container but not to release, the item is removed in origin container. When I release it, I didn't add `(cdkDropListDropped)`, so the item is recovered. But what I want is the item can into another container but not to be removed in any time. 

In summary, can you provide `(cdkDragLeaved)` to determine how to deal with the item when the item into another container?
 
#### Use Case

Just imaging you are ordering foods from menu in the restaurant. You need to drag the food into your shopping cart. But the item of the menu should't be removed.
![image](https://user-images.githubusercontent.com/30228406/89366962-7e7d3100-d70a-11ea-8ae0-c7057039dc66.png)

**Comment by devversion** [comment]
Hi @Mr-Eve. great issue! thanks. I think what you are trying to achieve should be partially possible already by just not using the move or transfer utilities the drag-drop provides, but rather by "copying" the element. For example:

```ts
    if (event.previousContainer.id === 'menu') {
      event.container.data.splice(event.currentIndex, 0,
        event.previousContainer.data[event.previousIndex]);
    }
```

The only problem I see here is that the drag placeholder from the menu list is temporarily removed when the item is being dragged over the `Shopping cart` drop list. cc. @crisbeto

**Comment by Mr-Eve** [comment]
Sorry, I'm not clear your meaning. The problem is , I have to use the way of drag to pick up the item of menu. I can't determine how to deal with the item when it into another container. The item is temporarily removed is default action.

**Comment by blankplanet** [comment]
Same Problem, When multiple cdkDropList exist, and those list are connected each other with droplist group or list connection.
After starting drag on source-list it generate placeholder, and when dragging item enter into destination-list,
placeholder also move to destination-list.

how can prevent placeholder disappear in source-list while dragging over other connected-list?

**Comment by blankplanet** [comment]
When multiple cdkDropList exist, and those list are connected each other with droplist group or list connection.
After starting drag on source-list it generate placeholder, and when dragging item enter into destination-list,
placeholder also move to destination-list.

how can prevent placeholder disappear in source-list while dragging over other connected-list?
there are some messy way that save placeholder element and insert it after exit source-list,
but i hope there are a new way like stop list DOM manipulation by cdk when drag exited list.

**Comment by Mr-Eve** [comment]
I solved it by a hack way. I think there is a long time during the CDK add this feature, if your really nedd it, you can try below code

    
    

    
    <ul
      cdkDropList
      cdkDropListSortingDisabled
      (cdkDropListEntered)="deleteFakeData()"
      (cdkDropListExited)="addFakeData($event)">
      
      <li cdkDrag [cdkDragData]="1">something1...</li>
      <li cdkDrag [cdkDragData]="2">something2...</li>
      <li cdkDrag [cdkDragData]="3">something3...</li>
    </ul>

    
    <ul
      cdkDropList
      (cdkDropListDropped)="dropAndDeleteFakeData($event)">
      <li>something1...</li>
      <li>something2...</li>
      <li>something3...</li>
    </ul>

**Comment by andreElrico** [comment]
@Mr-Eve is what you want a copy feature? somethink like here? 

https://jsbin.com/bifuyab/edit?js,output

**Comment by Mr-Eve** [comment]
@andreElrico yes, but it looks not under angular enviroment?

**Comment by timonkrebs** [comment]
I also would like to have this feature.

**Comment by Eve-Sama** [comment]
#13906

**Comment by Eve-Sama** [comment]
It's almost a year.

**Comment by jdpnielsen** [comment]
Why close this issue?
